app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -355,7 +365,68 @@ private void loadXml(BufferedReader in, ContentHandler parser) {
 			throw new LoadException("XML parsing problem: " + e);
 		}
 	}
-	
+	private void loadJson(BufferedReader in) {
+
+		StringBuffer buffer=new StringBuffer();
+		SimpleDateFormat df=new SimpleDateFormat();
+		 HashMap<String,Schedule.Line> tentMap=new HashMap<String,Schedule.Line>();
+		Scanner s=new Scanner(in);
+		while (s.hasNext()){
+			buffer.append(s.nextLine());
+
+		}
+		String output=buffer.toString();
+		Schedule.Line line=null;
+		try {
+			JSONArray events=new JSONArray(output);
+			for(int i=0;i<events.length();i++){
+				JSONObject event=events.getJSONObject(i);
+				String uid=event.getString("id");
+				String title=event.getString("title");
+				String startTimeS=event.getString("start_time");
+				String endTimeS=event.getString("end_time");
+				Date startTime,endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item=new Schedule.Item(uid,title,startTime,endTime);
+				item.setDescription(event.getString("long_abstract"));
+				if(event.getString("signup_url")!="null"&&event.getString("signup_url")!=null){
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+				JSONObject microlocation=event.getJSONObject("microlocation");
+				String location=microlocation.getString("name");
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+				line.addItem(item);
+			}
+
+		} catch (JSONException e) {
+			e.printStackTrace();
```

Comment:
Not that Giggity has flawless error handling/reporting as it is, please do make sure this one gets reported to the user. :-)